Problems with sub datasets in a jrxml file (ClassNotFoundExeception)

Using a simple jrxml file with a standart JSON Datasource works without any issues.
The project is setup like in the documentation and works without any issues even with resourcebundles(language files) with the 1.0.2 release inside an multistage docker container.

Using an report, which contains sub datasets leads to a ClassNotFoundException when filling the report.

it seems jasperreports creates some innerclasses for this kind of sub datasets and the current classloader is not able to find this class:

(FixedPriceGoAEInvoice.jrxml is the design file, containing the sub dataset "InvoicePositionsDataset")
Additional Notes: this kind of report works without any issues in DevMode, the problem happens only inside native Mode.

```
java.lang.RuntimeException: java.lang.ClassNotFoundException: FixedPriceGoAEInvoice_InvoicePositionsDataset_4ffb55f9b3fdec22f0d1299da07c2208f610dc0af680922cf49a5a2163f2ed4f
	at net.sf.jasperreports.engine.util.JRClassLoader.loadClass(JRClassLoader.java:29)
	at net.sf.jasperreports.engine.util.CompiledClassesLoader.loadCompiledClass(CompiledClassesLoader.java:87)
	at net.sf.jasperreports.engine.util.JRClassLoader.loadClassFromBytes(JRClassLoader.java:286)
	at net.sf.jasperreports.engine.design.JRAbstractJavaCompiler.loadClass(JRAbstractJavaCompiler.java:168)
	at net.sf.jasperreports.engine.design.JRAbstractJavaCompiler.loadEvaluator(JRAbstractJavaCompiler.java:117)
	at net.sf.jasperreports.engine.design.JRAbstractCompiler.createEvaluator(JRAbstractCompiler.java:459)
	at net.sf.jasperreports.engine.design.JRAbstractCompiler.loadEvaluator(JRAbstractCompiler.java:427)
	at net.sf.jasperreports.engine.JasperCompileManager.getEvaluator(JasperCompileManager.java:374)
	at net.sf.jasperreports.engine.fill.JRFillDataset.createCalculator(JRFillDataset.java:526)
	at net.sf.jasperreports.engine.fill.JRFillDataset.createCalculator(JRFillDataset.java:516)
	at net.sf.jasperreports.engine.fill.BaseReportFiller.createDatasets(BaseReportFiller.java:227)
	at net.sf.jasperreports.engine.fill.BaseReportFiller.<init>(BaseReportFiller.java:181)
	at net.sf.jasperreports.engine.fill.JRBaseFiller.<init>(JRBaseFiller.java:273)
	at net.sf.jasperreports.engine.fill.JRVerticalFiller.<init>(JRVerticalFiller.java:82)
	at net.sf.jasperreports.engine.fill.JRFiller.createBandReportFiller(JRFiller.java:252)
	at net.sf.jasperreports.engine.fill.JRFiller.createReportFiller(JRFiller.java:272)
	at net.sf.jasperreports.engine.fill.JRFiller.fill(JRFiller.java:157)
	at net.sf.jasperreports.engine.JasperFillManager.fillFromRepo(JasperFillManager.java:671)
	.....
       <snip>
        ......
	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:627)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@21.0.5/java.lang.Thread.runWith(Thread.java:1596)
	at java.base@21.0.5/java.lang.Thread.run(Thread.java:1583)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)
Caused by: java.lang.ClassNotFoundException: FixedPriceGoAEInvoice_InvoicePositionsDataset_4ffb55f9b3fdec22f0d1299da07c2208f610dc0af680922cf49a5a2163f2ed4f
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:122)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:86)
	at java.base@21.0.5/java.lang.Class.forName(DynamicHub.java:1359)
	at java.base@21.0.5/java.lang.Class.forName(DynamicHub.java:1348)
	at net.sf.jasperreports.engine.util.JRClassLoader.loadClass(JRClassLoader.java:27)
	... 44 more
```

any ideas how to solve this ?

Thanx in advance

--- a/deployment/src/main/java/io/quarkiverse/jasperreports/deployment/JasperReportsProcessor.java
+++ b/deployment/src/main/java/io/quarkiverse/jasperreports/deployment/JasperReportsProcessor.java
@@ -276,6 +276,7 @@ class JasperReportsProcessor extends AbstractJandexProcessor {
                 "javax.swing.text.rtf",
                 "sun.datatransfer",
                 "sun.swing",
+                "sun.lwawt.LWWindowPeer",
                 net.sf.jasperreports.components.headertoolbar.HeaderToolbarElement.class.getPackageName(),
                 net.sf.jasperreports.components.iconlabel.IconLabelElement.class.getPackageName(),
                 net.sf.jasperreports.components.list.UnusedSpaceImageRenderer.class.getName(),

--- a/pom.xml
+++ b/pom.xml
@@ -107,6 +107,11 @@
             </dependency>
             <dependency>
                 <groupId>net.sf.jasperreports</groupId>
+                <artifactId>jasperreports-functions</artifactId>
+                <version>${version.jasperreports}</version>
+            </dependency>
+            <dependency>
+                <groupId>net.sf.jasperreports</groupId>
                 <artifactId>jasperreports-jdt</artifactId>
                 <version>${version.jasperreports}</version>
             </dependency>

--- a/runtime/pom.xml
+++ b/runtime/pom.xml
@@ -60,6 +60,16 @@
         </dependency>
         <dependency>
             <groupId>net.sf.jasperreports</groupId>
+            <artifactId>jasperreports-functions</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>commons-logging</groupId>
+                    <artifactId>commons-logging</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>net.sf.jasperreports</groupId>
             <artifactId>jasperreports-jdt</artifactId>
             <exclusions>
                 <exclusion>

